test/functional/services/doc_table.js

```diff
@@ -75,6 +77,8 @@ export function DocTableProvider({ getService }) {
     async toggleRowExpanded(row) {
       const rowExpandToggle = await this.getRowExpandToggle(row);
       await rowExpandToggle.click();
+      // Not sure how to determine this rowExpandToggle completed but the next step often fails without this sleep
+      await PageObjects.common.sleep(2222);
```

Comment:
Why a sleep instead of wrapping the following line in a `retry.try`?

Reply:
I don't think the click is failing, I think we're trying to find `detailsRow` too quickly after clicking. Those rows can be a little slow to render after the caret is clicked. So I'm proposing we wrap line 83 in a retry, e.g.

```
retry.try(async () => {
  return row.findByXpath('./following-sibling::*[@data-test-subj="docTableDetailsRow"]');
});
```

This would allow the test to wait a moment for the row to render, without necessarily having to wait a full two seconds, correct?

Reply:
You could push this into the find service with a new call `find.descendantByXpath(selector, parentElement)` . This should use then our best practices for the retry and making sure the element isn't stale.